relayer/src/erc20/mod.rs

```diff
@@ -0,0 +1,334 @@
+use super::{ethereum_checkpoints::utils, *};
+use alloy::{
+    network::primitives::BlockTransactionsKind,
+    providers::{Provider, ProviderBuilder},
+    rpc::types::{Log, Receipt, ReceiptEnvelope, ReceiptWithBloom},
+};
+use alloy_consensus::TxType;
+use alloy_eips::BlockNumberOrTag;
+use alloy_primitives::Log as PrimitiveLog;
+use alloy_rlp::Encodable;
+use anyhow::{anyhow, Result as AnyResult};
+use checkpoint_light_client_io::ethereum_common::{
+    beacon::{self, light::Block as LightBeaconBlock, Block as BeaconBlock},
+    memory_db,
+    patricia_trie::{TrieDB, TrieDBMut},
+    trie_db::{Recorder, Trie, TrieMut},
+    utils as eth_utils, H256, SLOTS_PER_EPOCH,
+};
+use erc20_relay_client::{
+    traits::Erc20Relay as _, Block, BlockBody, BlockHeader, BlockInclusionProof, Erc20Relay,
+    EthToVaraEvent, ExecutionPayload,
+};
+use futures::StreamExt;
+use gclient::{GearApi, WSAddress};
+use reqwest::Client;
+use sails_rs::{calls::*, events::*, gclient::calls::*, prelude::*};
+use std::cmp::Ordering;
+
+fn into_idl_execution_payload(
```

Comment:
Is it due to the fact that `sails` cannot encode it automatically?

Reply:
Nope, it is consequence of regenerating types by sails. I requested the corresponding feature some time ago - https://github.com/gear-tech/sails/pull/538 . It is implemented however there is an issue - https://github.com/gear-tech/sails/issues/556 .

So for now we have to use these mapping funcs

Reply:
ok, then just open an issue in `gear-bridges` and link everything there